### PR TITLE
Congrats: Refactored domain bulk transfer congrats page

### PR DESCRIPTION
**Pull request by wongasy**
Related to https://github.com/Automattic/dotcom-forge/issues/5121

## Proposed Changes

This PR refactors the Bulk Transfer Congrats Page to adapt the new approach for unifying the congratulation pages using the `<ThankYouV2>` component. See pb5gDS-38y-p2#comment-4475 for details.

## Testing Instructions

* If you don't have a domain previously purchased, purchase one.
* If you have a domain handy, get the respective receipt ID using the following steps:
    * Go to Upgrades > Purchases > Billing History
    * Look for the domain purchase and click on "View receipt" and you should be able to find the receipt ID
* Point `public-api.wordpress.com` to your sandbox IP.
* Apply the following diff 332e2-pb/#diff to your sandbox to force the purchase to be returned as a domain transfer:
* Visit `/checkout/thank-you/:site/:receipt_id` and assert that you see the Domain Transfer Page.
* Compare the same page in production and assert that you see the same page content, with the exception of the following:
    * Spacing is a bit different, since we migrated to the unified template.
    * ~The confetti animation is removed.~ [**UPDATE**: We are keeping the animation after all p9Jlb4-7Hx-p2#comment-10513] 
* Assert that all buttons and links click as expected.

Screen Shot 2024-01-11 at 4 34 05 PM

* Apply the following diff 332e3-pb/#diff to your sandbox to force a purchase of `$0`:
* Visit `/checkout/thank-you/:site/:receipt_id` and assert that you see the Domain Transfer Page and that it functions the same way as the production page with the same remarks noted above regarding spacing / confetti animation.

Screen Shot 2024-01-11 at 4 42 08 PM

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-bulk-transfer.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				onClick={ () => handleUserClick( '/setup/domain-transfer' ) }
+				className="is-secondary"
+			>
+				{ __( 'Transfer more domains' ) }
+			</Button>
+
+			<Button
+				href="/domains/manage"
+				className="manage-all-domains"
+				onClick={ () => handleUserClick( '/domains/manage' ) }
```
Comment (fredrikekelund) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-bulk-transfer.tsx]:
```suggestion
				href={ domainManagementRoot() }
				className="manage-all-domains"
				onClick={ () => handleUserClick( domainManagementRoot() ) }
```
`domainManagementRoot` is exported from `client/my-sites/domains/paths.js`

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-redesign-v2.js [hunk 2]
```diff
@@ -44,6 +45,18 @@ describe( 'isRedesignV2', () => {
 		expect( isRedesignV2( props ) ).toBe( false );
 	} );
 
+	it( 'should return false if there purchases contain only domain transfers', () => {
```
Comment (fredrikekelund) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-redesign-v2.js]:
```suggestion
	it( 'should return false if the purchases contain only domain transfers', () => {
```

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-refactored-for-thank-you-v2.js [hunk 3]
```diff
@@ -31,6 +32,18 @@ describe( 'isRefactoredForThankYouV2', () => {
 		expect( isRefactoredForThankYouV2( props ) ).toBe( true );
 	} );
 
+	it( 'should return true if there purchases contain only domain transfers', () => {
```
Comment (fredrikekelund) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/test/is-refactored-for-thank-you-v2.js]:
```suggestion
	it( 'should return true if the purchases contain only domain transfers', () => {
```

**Review by fredrikekelund — APPROVED** [review]
Excellent work planning out this refactoring! This LGTM 👍

I really appreciated reviewing this to understand the plans that you laid out on P2 better.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-bulk-transfer.tsx (hunk 1 above)
Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/domain-bulk-transfer.tsx]: Nice! Thanks for pointing to the function :)

**Comment by wongasy** [comment]
A quick note that we're keeping the confetti animation after all, see p9Jlb4-7Hx-p2#comment-10513. Update made here: https://github.com/Automattic/wp-calypso/pull/86203/commits/538dd4e9a3a6af82c43979dac4484e0539b3be72

**Pull request merged**